linera-rpc/src/grpc/conversions.rs

```diff
@@ -559,15 +559,21 @@ impl TryFrom<api::CryptoHash> for CryptoHash {
     }
 }
 
-impl From<BlobContent> for api::BlobContent {
-    fn from(blob: BlobContent) -> Self {
-        Self { bytes: blob.bytes }
+impl TryFrom<BlobContent> for api::BlobContent {
+    type Error = GrpcProtoConversionError;
+
+    fn try_from(blob: BlobContent) -> Result<Self, Self::Error> {
+        Ok(Self {
+            bytes: bincode::serialize(&blob)?,
```

Comment:
Now we're exposing the different blob types to contracts after all.

Ideally the API shouldn't treat bytecode, application descriptions and binary data the same at all. From a user perspective they have nothing to do with each other.